include/cuco/detail/static_multimap/device_view_impl.inl

```diff
@@ -1002,6 +1002,7 @@ class static_multimap<Key, Value, Scope, Allocator, ProbeSequence>::device_view_
           flushing_cg, *flushing_cg_counter, output_buffer, num_matches, output_begin);
         // First lane reset warp-level counter
         if (flushing_cg.thread_rank() == 0) { *flushing_cg_counter = 0; }
+        flushing_cg.sync();
```

Comment:
Note also that the `atomicAdd` is on lane-zero of `probing_cg` whereas this write is on lane-zero of `flushing_cg` (so those _might_ be different threads).

Reply:
> but with an atomicAdd (does that provide an ordering guarantee wrt the write?).

`atomicAdd` has [relaxed memory order](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#atomic-functions) so non-atomic reads/writes to the same location can migrate past the atomic read/write I believe. I guess this is why it is better to use `atomic_ref` because that protects a variable, not a location.